beacon-chain/blockchain/process_block.go

```diff
@@ -250,51 +238,6 @@ func (s *Service) onBlockBatch(ctx context.Context, blks []*ethpb.SignedBeaconBl
 	return preState, fCheckpoints, jCheckpoints, nil
 }
 
-// handles the state post transition and saves the appropriate checkpoints and forkchoice
-// data.
-func (s *Service) handlePostStateInSync(ctx context.Context, signed *ethpb.SignedBeaconBlock,
-	blockRoot [32]byte, postState *stateTrie.BeaconState) error {
-
-	b := signed.Block
-
-	s.saveInitSyncBlock(blockRoot, signed)
-
-	if err := s.stateGen.SaveState(ctx, blockRoot, postState); err != nil {
-		return errors.Wrap(err, "could not save state")
-	}
-
-	if err := s.insertBlockAndAttestationsToForkChoiceStore(ctx, b, blockRoot, postState); err != nil {
-		return errors.Wrapf(err, "could not insert block %d to fork choice store", b.Slot)
-	}
-
-	// Rate limit how many blocks (2 epochs worth of blocks) a node keeps in the memory.
-	if uint64(len(s.getInitSyncBlocks())) > initialSyncBlockCacheSize {
-		if err := s.beaconDB.SaveBlocks(ctx, s.getInitSyncBlocks()); err != nil {
-			return err
-		}
-		s.clearInitSyncBlocks()
-	}
-
-	// Update finalized check point. Prune the block cache and helper caches on every new finalized epoch.
-	if postState.FinalizedCheckpointEpoch() > s.finalizedCheckpt.Epoch {
-		if err := s.beaconDB.SaveBlocks(ctx, s.getInitSyncBlocks()); err != nil {
-			return err
-		}
-		s.clearInitSyncBlocks()
-
-		if err := s.updateFinalized(ctx, postState.FinalizedCheckpoint()); err != nil {
-			return err
-		}
-
-		fRoot := bytesutil.ToBytes32(postState.FinalizedCheckpoint().Root)
-		if err := s.stateGen.MigrateToCold(ctx, fRoot); err != nil {
```

Comment:
what is the reason for `migrateToCold` not being required now ?

Reply:
It's still there, we moved it within `updatedFinalized`
https://github.com/prysmaticlabs/prysm/pull/6562/files#diff-f407600862245d51b706766ae42ee345R208